src/batch.jl

```diff
@@ -0,0 +1,191 @@
+"""
+    BatchIntegrand(f!, y::AbstractVector, x::AbstractVector, max_batch=typemax(Int))
```

Comment:
Maybe add an outer constructor where `max_batch` is a keyword, to match your other outer constructor.